pkg/autoscaler/health.go

```diff
@@ -0,0 +1,61 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package autoscaler
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+
+	"github.com/golang/glog"
+)
+
+type healthInfo struct {
+	m   *sync.RWMutex
+	err error
+}
+
+func newHealthInfo() *healthInfo {
+	return &healthInfo{m: &sync.RWMutex{}, err: nil}
+}
+
+func (h *healthInfo) setError(err error) {
+	h.m.Lock()
+	h.err = err
+	h.m.Unlock()
+}
+
+func (h *healthInfo) getError() error {
+	h.m.RLock()
+	defer h.m.RUnlock()
```

Comment:
here you use defer.  Above you use explicit unlock.  Prefer defer forconsistency?

Reply:
using defer